test/test_transforms_tensor.py

```diff
@@ -97,121 +97,23 @@ def _test_op(func, method, device, fn_kwargs=None, meth_kwargs=None, test_exact_
-        batch_tensors = torch.rand(4, 3, 44, 56, device=self.device)
-
-        test_configs = [
-            {"value": 0.2},
-            {"value": "random"},
-            {"value": (0.2, 0.2, 0.2)},
-            {"value": "random", "ratio": (0.1, 0.2)},
-        ]
-
-        for config in test_configs:
-            fn = T.RandomErasing(**config)
-            scripted_fn = torch.jit.script(fn)
-            _test_transform_vs_scripted(fn, scripted_fn, tensor)
-            _test_transform_vs_scripted_on_batch(fn, scripted_fn, batch_tensors)
-
-        with get_tmp_dir() as tmp_dir:
-            scripted_fn.save(os.path.join(tmp_dir, "t_random_erasing.pt"))
-
-    def test_convert_image_dtype(self):
-        tensor, _ = _create_data(26, 34, device=self.device)
-        batch_tensors = torch.rand(4, 3, 44, 56, device=self.device)
-
-        for in_dtype in int_dtypes() + float_dtypes():
-            in_tensor = tensor.to(in_dtype)
-            in_batch_tensors = batch_tensors.to(in_dtype)
-            for out_dtype in int_dtypes() + float_dtypes():
-
-                fn = T.ConvertImageDtype(dtype=out_dtype)
-                scripted_fn = torch.jit.script(fn)
-
-                if (in_dtype == torch.float32 and out_dtype in (torch.int32, torch.int64)) or \
-                        (in_dtype == torch.float64 and out_dtype == torch.int64):
-                    with self.assertRaisesRegex(RuntimeError, r"cannot be performed safely"):
-                        _test_transform_vs_scripted(fn, scripted_fn, in_tensor)
-                    with self.assertRaisesRegex(RuntimeError, r"cannot be performed safely"):
-                        _test_transform_vs_scripted_on_batch(fn, scripted_fn, in_batch_tensors)
-                    continue
-
-                _test_transform_vs_scripted(fn, scripted_fn, in_tensor)
-                _test_transform_vs_scripted_on_batch(fn, scripted_fn, in_batch_tensors)
-
-        with get_tmp_dir() as tmp_dir:
-            scripted_fn.save(os.path.join(tmp_dir, "t_convert_dtype.pt"))
-
-    def test_autoaugment(self):
-        tensor = torch.randint(0, 256, size=(3, 44, 56), dtype=torch.uint8, device=self.device)
-        batch_tensors = torch.randint(0, 256, size=(4, 3, 44, 56), dtype=torch.uint8, device=self.device)
-
-        s_transform = None
-        for policy in T.AutoAugmentPolicy:
-            for fill in [None, 85, (10, -10, 10), 0.7, [0.0, 0.0, 0.0], [1, ], 1]:
-                transform = T.AutoAugment(policy=policy, fill=fill)
-                s_transform = torch.jit.script(transform)
-                for _ in range(25):
-                    _test_transform_vs_scripted(transform, s_transform, tensor)
-                    _test_transform_vs_scripted_on_batch(transform, s_transform, batch_tensors)
-
-        if s_transform is not None:
-            with get_tmp_dir() as tmp_dir:
-                s_transform.save(os.path.join(tmp_dir, "t_autoaugment.pt"))
+@cpu_only
```

Comment:
This test is actually called twice: with self.device = 'cpu' and with self.device = 'cuda' from the CUDATester class.

So instead of using the `@cpu_only` decorator, we should parametrize with a new

```py
@pytest.mark.parametrize('device', cpu_and_gpu())
```

and replace `self.device` by device.

You'll need to remove `device` from the parametrization below as well :)